ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -1744,6 +1747,15 @@ public void setForwardedAddresses(List<String> forwardedAddresses) {
   public String getReloadableAuxJars() {
     return StringUtils.join(preReloadableAuxJars, ',');
   }
+
+  public void updateProgressMonitor(ProgressMonitor progressMonitor) {
```

Comment:
Since access to Session state Itself is via thread local multiple threads will not have access to this apart from background thread and foreground thread on HS2. Adding the synchronization will add unnecessary overhead i think.